### PR TITLE
Add NTEE Org filter to map home page

### LINKED ISSUE #9: Map should allow filtering on the "nonprofit type" field
Dropdown with all available options

@designmatty  assuming this should live as a control on the map?

### PR DESCRIPTION
**Pull request by CloudNiner**
## Overview

Adds filter control to map home page that allows user to filter map by org type.

## Demo

_Updated 11/8/2017:_

screen shot 2017-11-08 at 14 48 42
screen shot 2017-11-08 at 14 47 48

## Notes

~~Pending clarification on whether this should also be added to the detail view~~This was requested, PR updated.

## Testing

Filter some stuff on the home page map.

Closes #9

### CODE REVIEW

File: app/scripts/map/filter-control-directive.js [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+ */
+
+(function () {
+    'use strict';
+
+    /** @ngInject */
+    function FilterController($scope, Config) {
+        var layer = null;
+        var filterColumn = 'ntee_org_type_new';
+        var noneName = 'None';
```
Comment (KlaasH) [inline app/scripts/map/filter-control-directive.js]: I think "All" would be better.

File: app/scripts/map/filter-control-directive.js [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+ *
+ * imls:area-analysis-control:draw:start Triggered when the user clicks on any of the custom
+ *                                       draw buttons
+ *     {String} drawType The button selected, currently one of 'circle'|'polygon'
+ *
+ * imls:area-analysis-control:draw:complete Triggered when the user finishes drawing a custom shape
+ *     {Event} drawEvent The event fired by the L.Draw control, passed through
+ *
+ * imls:area-analysis-control:draw:cancel Listened for on scope, trigger elsewhere to clear the
+ *                                        draw handler and reset the control
```
Comment (KlaasH) [inline app/scripts/map/filter-control-directive.js]: Stale copy-pasta documentation.

**Review by KlaasH — APPROVED** [review]
Looks good.  A couple little notes below, then :+1:.

Re "Pending clarification on whether this should also be added to the detail view", I'll say that having the filter reset when you view details for an org looks somewhat weird.

File: app/scripts/map/filter-control-directive.js (hunk 1 above)
Comment (CloudNiner) [reply app/scripts/map/filter-control-directive.js]: Way better.

**Comment by CloudNiner** [comment]
@KlaasH feel free to take another look, otherwise will merge tomorrow am.

**Comment by nevillevak** [comment]
@CloudNiner - This looks good. Thanks!

**Pull request merged**